Purchases/Purchasing/StoreKitAbstractions/StoreProduct.swift

```diff
@@ -127,6 +125,16 @@ public typealias SK2Product = StoreKit.Product
         return SubscriptionPeriod.from(sk2SubscriptionPeriod: skSubscriptionPeriod)
     }
 
+    @objc public override var introductoryPrice: PromotionalOffer? {
+        self.underlyingSK2Product.subscription?.introductoryOffer
+            .map(PromotionalOffer.init)
+    }
+
+    @objc public override var discounts: [PromotionalOffer] {
+        (self.underlyingSK2Product.subscription?.promotionalOffers ?? [])
+            .compactMap(PromotionalOffer.init)
```

Comment:
is `compact` here needed? the values inside the array should never be nil, right?

Reply:
But the initializer returns optional.

Reply:
ohh I'd missed that 👌